src/Form/Extension/AddToCartTypeExtension.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Form\Extension;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusGiftCardPlugin\Factory\GiftCardFactoryInterface;
+use Setono\SyliusGiftCardPlugin\Form\Type\AddToCartGiftCardInformationType;
+use Setono\SyliusGiftCardPlugin\Model\OrderItemUnitInterface;
+use Setono\SyliusGiftCardPlugin\Model\ProductInterface;
+use Setono\SyliusGiftCardPlugin\Order\AddToCartCommandInterface;
+use Sylius\Bundle\CoreBundle\Form\Type\Order\AddToCartType;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+final class AddToCartTypeExtension extends AbstractTypeExtension
+{
+    private GiftCardFactoryInterface $giftCardFactory;
+
+    private EntityManagerInterface $giftCardManager;
+
+    public function __construct(
+        GiftCardFactoryInterface $giftCardFactory,
+        EntityManagerInterface $giftCardManager
+    ) {
+        $this->giftCardFactory = $giftCardFactory;
+        $this->giftCardManager = $giftCardManager;
+    }
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [
+            AddToCartType::class,
+        ];
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, [$this, 'reworkFormForGiftCard']);
+
+        $builder->addEventListener(FormEvents::POST_SUBMIT, [$this, 'populateCartItem']);
+    }
+
+    public function reworkFormForGiftCard(FormEvent $event): void
+    {
+        /** @var AddToCartCommandInterface|null $data */
+        $data = $event->getData();
+        if (null === $data) {
+            return;
+        }
+
+        /** @var ProductInterface|null $product */
+        $product = $data->getCartItem()->getProduct();
+        if (null === $product) {
+            return;
+        }
+
+        // If the product is a gift card, we remove the quantity and add the GiftCardInformation fields
+        if ($product->isGiftCard()) {
+            $form = $event->getForm();
+            $form->get('cartItem')->remove('quantity');
```

Comment:
What if I want to buy 10 gift cards for my friends or something? :)

Reply:
With the same custom message ? It would be quite weird IMO, and a bit of an edge case.

But perhaps we can keep the quantity input. WDYT ?

Reply:
I don't have to input the custom message do I? But yes, I think the qty field should stay